source/server-selection/server-selection.rst

```diff
@@ -1041,12 +1043,57 @@ The shortest average round trip time (RTT) from among suitable servers anchors
 one end of the latency window (``A``).  The other end is determined by adding
 ``localThresholdMS`` (``B = A + localThresholdMS``).
 
-A server MUST be selected randomly from among suitable servers that have an
-average RTT (``RTT``) within the latency window (i.e. ``A ≤ RTT ≤ B``).
+A server MUST be selected from among suitable servers that have an average RTT
+(``RTT``) within the latency window (i.e. ``A ≤ RTT ≤ B``). In other words, the
+suitable server with the shortest average RTT is **always** a possible choice.
+Other servers could be chosen if their average RTTs are no more than
+``localThresholdMS`` more than the shortest average RTT.
+
+If a driver does not implement a CMAP-compliant connection pool, then a server
+MUST be selected at random from those within the latency window.
+
+If a driver does implement a CMAP-compliant connection pool, then a server MUST
+be selected according to the following algorithm, which is a slight modification
+of the `"Power of Two Random Choices with Min Connections"
+<https://www.nginx.com/blog/nginx-power-of-two-choices-load-balancing-algorithm/>`_
+load balancing algorithm. The steps are as follows:
+
+1. Select two servers at random from the set of available servers in the latency
+   window. If there is only 1 server in the latency window, just return that server
+   without proceeding to the next steps.
+
+2. Calculate each server's ``operationCount``, which is the sum of the pool's
+   activeConnectionCount and the length of the pool's WaitQueue. If the
+   difference between the two server's ``operationCounts`` is greater than 5% of
+   maxPoolSize, then choose the server with the smaller ``operationCount``.
+
+3. Otherwise, choose the server with the higher ``availableConnectionCount``.
```

Comment:
This should also say what happens if availableConnectionCount are the same.

Reply:
We decided to omit the consideration of availableConnectionCounts altogether to simplify the algorithm and remove the need for server selection to have a dependency on CMAP.